target/linux/lantiq/files/arch/mips/boot/dts/lantiq/ar10_tplink_archer-d2.dts

```diff
@@ -0,0 +1,268 @@
+ 	model = "TP-Link Archer D2 v1";
+
+ 	memory@0 {
+ 		device_type = "memory";
+ 		reg = <0x0 0x4000000>;
+ 	};
+
+	aliases {
+                led-boot = &led_power;
+                led-failsafe = &led_power;
+                led-running = &led_power;
+                led-upgrade = &led_power;
+
+                led-dsl = &led_dsl;
+                led-internet = &led_internet;
+                led-wifi = &led_wlan;
+        };
+
+	keys: keys {
+                compatible = "gpio-keys-polled";
+                poll-interval = <100>;
+
+                reset {
+                        label = "reset";
+                        gpios = <&gpio 34 GPIO_ACTIVE_LOW>;
+                        linux,code = <KEY_RESTART>;
+                };
+
+                wifi {
+                        label = "wifi";
+                        gpios = <&gpio 9 GPIO_ACTIVE_HIGH>;
+                        linux,code = <KEY_RFKILL>;
+                        linux,input-type = <EV_SW>;
+                };
+
+                wps {
+                        label = "wps";
+                        gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+                        linux,code = <KEY_WPS_BUTTON>;
+                };
+        };
+
+        leds: leds {
+                compatible = "gpio-leds";
+
+                led_power: power {
+                        label = "power";
+                        gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+                        default-state = "keep";
+                };
+
+                led_dsl: dsl {
+                        label = "dsl";
+                        gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+                };
+
+                led_internet: internet {
+                        label = "internet";
+                        gpios = <&gpio 5 GPIO_ACTIVE_LOW>;
+                };
+
+                usb {
+                        label = "usb";
+                        gpios = <&gpio 26 GPIO_ACTIVE_LOW>;
+                        trigger-sources = <&ehci_port1>;
+                        linux,default-trigger = "usbport";
+                };
+
+                lan {
+                        label = "lan";
+                        gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+                };
+
+                led_wlan: wlan {
+                        label = "wlan";
+                        gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+                };
+
+                wps {
+                        label = "wps";
+                        gpios = <&gpio 27 GPIO_ACTIVE_LOW>;
+                };
+        };
+
+        usb_vbus: regulator-usb-vbus {
+		compatible = "regulator-fixed";
+
+		regulator-name = "USB_VBUS";
+
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+
+		gpio = <&gpio 25 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+};
+
+&spi {
+        status = "okay";
+
+        flash@0 {
+                compatible = "jedec,spi-nor";
+                reg = <1>;
+                spi-max-frequency = <25000000>;
```

Comment:
Can it go faster?

Reply:
I don't know, but other devices with the same flash chip (https://openwrt.org/toh/qxwlan/e600g) had it set to this frequency